registry/api/v2/urls.go

```diff
@@ -175,6 +176,8 @@ func (ub *URLBuilder) BuildManifestURL(ref reference.Named) (string, error) {
 		tagOrDigest = v.Tag()
 	case reference.Digested:
 		tagOrDigest = v.Digest().String()
+	default:
+		return "", fmt.Errorf("Named Reference must have tag or digest")
```

Comment:
Error messages should generally start with a lowercase character.

I'd suggest `reference must have a tag or digest`.